spacy/cli/debug_model.py

```diff
@@ -77,7 +77,7 @@ def debug_model_cli(
             exits=1,
         )
     model = pipe.model
-    debug_model(C, nlp, model, print_settings=print_settings)
+    debug_model(T, nlp, model, print_settings=print_settings)
```

Comment:
I think this edit needs to be reverted: `debug_model` assumes it gets the full config, and it needs it, because the training corpus can be defined outside the `training` block (looked up with `dot_to_object`)

Reply:
Ahh, we should use the new `resolve_dot_names` helper here, which will walk through the config and resolve only the relevant referenced sections. See `loop.train` for an example.

(With the new Thinc, we're now able to resolve individual sections more effectively so we only want to be creating what we actually need.)

Reply:
Sure, but that doesn't change the fact that `debug_model` needs access to `C`, and not just the `T` section, right?

Reply:
Yeah, the `resolve_dot_names` would just receive the full (unresolved) config and would then take care of that itself, depending on which sections are needed and where the corpus is defined.

Reply:
Also just noticed there's something we need to fix in the `resolve_dot_names`, so I'm doing that now